experimental/dispatch_profiler/batch_matmul.py

```diff
@@ -0,0 +1,230 @@
+
+  def __init__(self):
+    self.mlir_dialect = MlirDialect.Linalg
+
+    self.linalg_row_row_matmul_template = """
+// Dispatch linalg.batch_matmul row-row layout 
+func.func @${operation_name}_${compilation_info_name}(
+  %lhs: tensor<${batch_count}x${problem_m}x${problem_k}x${datatype_lhs}>,
+  %rhs: tensor<${batch_count}x${problem_k}x${problem_n}x${datatype_rhs}>) -> tensor<${batch_count}x${problem_m}x${problem_n}x${datatype_result}>
+{
+  %c0 = arith.constant 0.0 : ${datatype_result}
+  %init = tensor.empty() : tensor<${batch_count}x${problem_m}x${problem_n}x${datatype_result}>
+  %inital_result = linalg.fill ins(%c0 : ${datatype_result}) outs(%init : tensor<${batch_count}x${problem_m}x${problem_n}x${datatype_result}>) -> tensor<${batch_count}x${problem_m}x${problem_n}x${datatype_result}>
+  %result = linalg.batch_matmul ${compilation_info_attribute} 
+                     ins(%lhs, %rhs: tensor<${batch_count}x${problem_m}x${problem_k}x${datatype_lhs}>, tensor<${batch_count}x${problem_k}x${problem_n}x${datatype_rhs}>)
+                     outs(%inital_result: tensor<${batch_count}x${problem_m}x${problem_n}x${datatype_result}>) -> tensor<${batch_count}x${problem_m}x${problem_n}x${datatype_result}>
+  return %result : tensor<${batch_count}x${problem_m}x${problem_n}x${datatype_result}>
+}
+"""
+
+  def emit(self, dispatch):
+    """Emit the matmul operation in the MLIR dialect for a single compilation info"""
+    compilation_info_attribute_template = """{compilation_info = #${compilation_info_name}}"""
+    compilation_info_attribute_str = SubstituteTemplate(
+        compilation_info_attribute_template,
+        {'compilation_info_name': dispatch.configuration.name()})
+    compilation_info_attribute = compilation_info_attribute_str \
+      if dispatch.configuration.config_type != CompilationConfigType.Default else ""
+
+    values = {
+        'operation_name': dispatch.operation.name(),
+        'compilation_info_attribute': compilation_info_attribute,
+        'batch_count': str(dispatch.operation.batch_count),
+        'problem_m': str(dispatch.operation.M),
+        'problem_n': str(dispatch.operation.N),
+        'problem_k': str(dispatch.operation.K),
+        'datatype_lhs': DataTypeName[dispatch.operation.lhs.datatype],
+        'datatype_rhs': DataTypeName[dispatch.operation.rhs.datatype],
+        'datatype_result': DataTypeName[dispatch.operation.result.datatype],
+        'compilation_info_name': dispatch.configuration.name()
+    }
+
+    return SubstituteTemplate(self.linalg_row_row_matmul_template, values)
+
+
+class ReferenceBatchMatmulOp(ReferenceOpInterface):
+  """Reference implementation for the batch matmul operation in numpy."""
+
+  def __init__(self, bmm_operation, op_reference_cache_path, dist_lhs,
+               dist_rhs):
+    self.bmm_operation = bmm_operation
+    self.op_reference_cache_path = op_reference_cache_path
+
+    if not os.path.exists(self.op_reference_cache_path):
```

Comment:
I have made changes to address the above request [here](https://github.com/openxla/iree/pull/13396/commits/8369566466802e99e25c7065c4128a907ad6c177). 

@stellaraccident , @GMNGeoffrey , @ScottTodd , @MaheshRavishankar, @julianwa , Understanding of the project-wide guidelines for taking similar changes, particularly when it comes to deciding between alternatives with very fine trade-offs will be helpful. I have noticed that `os.path` is used in numerous IREE-specific Python scripts. It would be greatly appreciated if we could receive some project-wide guidance or recommendations to help us _prioritize_ similar changes effectively.